### PR TITLE
Fix/4237  same language translation

### PR DESCRIPTION
**Pull request by grignards**
See #4237.

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [x] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [x] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

### CODE REVIEW

File: backend/tracim_backend/lib/translate/providers.py [hunk 1]
```diff
@@ -37,15 +40,63 @@ def __init__(
...
                 api_key=self._config.TRANSLATION_SERVICE__SYSTRAN__API_KEY,
+                timeout=self._config.TRANSLATION_SERVICE__TIMEOUT,
             )
         elif self._config.TRANSLATION_SERVICE__PROVIDER == TranslationProvider.TEST:
             logger.warning(self, "Running in test translation service !")
             return TestTranslationService()
         else:
             raise TracimException("Translation Service not available")
+
+    def translate_raw_content(
```
Comment (inkhey) [inline backend/tracim_backend/lib/translate/providers.py:57]: I don't really understood the separation between translate_raw_content usage and view for now.
`get_html_document_translation` do not use this method and `filename` is modified both outside and inside the function which seems confusing for me.

**Review by inkhey — CHANGES_REQUESTED** [review]

File: backend/tracim_backend/lib/translate/providers.py (hunk 1 above)
Comment (grignards) [reply backend/tracim_backend/lib/translate/providers.py:57]: Ah… yes I forgot to use it in `get_html_document_translation`, this is fixed.

**Review by raphj — APPROVED** [review]

**Review by inkhey — APPROVED** [review]

**Review by PhilippeAccorsi — APPROVED** [review]

File: backend/development.ini.sample [hunk 2]
```diff
@@ -770,6 +770,13 @@ caldav.radicale.headers.Access-Control-Expose-Headers = Etag
 
 ; translation_service.enabled = False
 
+
+## Translation service timeout in seconds.
+# This timeout will be applied to each translation request.
+; translation_service.timeout = 0
+# example:
+# translation_service.timeout = 5
```
Comment (PhilippeAccorsi) [inline backend/development.ini.sample:778]: Maybe add bigger timeout value like 10-20s

I think 5sec is to slow for a timeout on this kind of feature.

Comment (grignards) [reply backend/development.ini.sample]: Done.

**Review by PhilippeAccorsi — APPROVED** [review]

**Pull request merged**